agg-ast/schema_derivation/src/schema_derivation.rs

```diff
@@ -773,22 +804,23 @@ impl DeriveSchema for TaggedOperator {
                 let var = m._as.clone();
                 let var = var.unwrap_or("this".to_string());
                 let input_schema = m.input.derive_schema(state)?;
-                let array_schema = match input_schema {
-                    Schema::Array(a) => *a,
-                    _ => {
-                        return Err(Error::InvalidExpressionForField(
-                            format!("{:?}", m.input),
-                            "input",
-                        ))
-                    }
-                };
+                if input_schema.satisfies(&NULLISH.clone()) == Satisfaction::Must {
```

Comment:
as mentioned in the macro above, these array operators in many cases can handle an array _or_ null, but we were treating them strictly as arrays, so these changes simply enable that behavior